esphome/components/http_request/ota/__init__.py

```diff
@@ -0,0 +1,195 @@
+import urllib.parse as urlparse
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.const import (
+    CONF_ESP8266_DISABLE_SSL_SUPPORT,
+    CONF_ID,
+    CONF_PASSWORD,
+    CONF_TIMEOUT,
+    CONF_URL,
+    CONF_USERNAME,
+)
+from esphome.components import esp32
+from esphome.components.ota import BASE_OTA_SCHEMA, ota_to_code, OTAComponent
+from esphome.core import CORE, coroutine_with_priority
+from .. import http_request_ns
+
+CODEOWNERS = ["@oarcher"]
+
+AUTO_LOAD = ["md5", "ota"]
```

Comment:
As this is an `ota` platform, it will always be loaded
```suggestion
AUTO_LOAD = ["md5"]
```